src/app/views/workflow-preview/WorkflowPreview.jsx

```diff
@@ -21,11 +22,25 @@ const propTypes = {
 export class WorkflowPreview extends Component {
     constructor(props) {
         super(props);
+        this.state = { cantabularDataset: false };
     }
 
-    handleBackButton = () => {
+    handleBackButton = async () => {
         const previousUrl = url.resolve("../");
-        this.props.dispatch(push(previousUrl));
+        await this.getDatasetType(this.props.params.datasetID);
+        if (this.state.cantabularDataset) {
+            this.props.dispatch(push(previousUrl + "/cantabular"));
+        } else {
+            this.props.dispatch(push(previousUrl));
+        }
+    };
+
+    getDatasetType = datasetID => {
+        return datasets.get(datasetID).then(response => {
+            if (response.next.type === "cantabular_table" || response.next.type === "cantabular_flexible_table") {
+                this.setState({ cantabularDataset: true });
```

Comment:
This doesn't allow you to falsify the flag, so perhaps this would be better:

```
const type = response.next.type;
this.setState({ cantabularDataset: type === "cantabular_table" || type === "cantabular_flexible_table" });
```

Reply:
Done :white_check_mark: